### PR TITLE
Viz - tooltip renovation (step #3)

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: js/renovation/viz/core/__tests__/tooltip.test.tsx [hunk 1]
```diff
@@ -271,11 +272,35 @@ describe('Render', () => {
     expect(tooltip.find('div').at(1).children().props()).toEqual({ valueText: 'Tooltip value text' });
     expect(tooltip.find('.tooltip-template').text()).toBe('Tooltip value text_template');
   });
+
+  it('should set on the div zIndex', () => {
+    const tooltip = shallow(<TooltipComponent {...{
+      ...props,
+      props: { ...props.props, zIndex: 3 },
+    } as any}
+    />);
```
Comment (MaximKudriavtsev) [inline js/renovation/viz/core/__tests__/tooltip.test.tsx]: The same with previous PR. It's hard to understand what does it mean:
```jsx
<TooltipComponent {...{
      ...props,
      props: { ...props.props, zIndex: 3 },
    } as any}
/>
```

File: js/renovation/viz/core/tooltip.tsx [hunk 2]
```diff
@@ -201,7 +205,17 @@ export class TooltipProps {
...
   @OneWay() interactive = false;
 
+  @OneWay() enabled = true;
+
+  @OneWay() shared = false;
+
+  @OneWay() location = 'center';
+
+  @OneWay() zIndex?: number;
+
```
Comment (MaximKudriavtsev) [inline js/renovation/viz/core/tooltip.tsx]: Why you did not specify a property type?

```ts
  @OneWay() enabled: boolean = true;

  @OneWay() shared: boolean = false;

  @OneWay() location: locationType = 'center';

  @OneWay() zIndex?: number;
```

File: js/renovation/viz/core/tooltip.tsx [hunk 3]
```diff
@@ -266,6 +282,21 @@ export class Tooltip extends JSXComponent(TooltipProps) {
...
   }
 
+  @Method()
+  isEnabled(): boolean {
+    return this.props.enabled;
+  }
+
+  @Method()
+  isShared(): boolean {
+    return !!this.props.shared;
```
Comment (MaximKudriavtsev) [inline js/renovation/viz/core/tooltip.tsx]: Is not `shared` a boolean type? Why do we use a `!!` operator?

File: js/renovation/viz/core/tooltip.tsx (hunk 2 above)
Comment (Krijovnick) [reply js/renovation/viz/core/tooltip.tsx]: The `shared` and `enabled` options have default values and it is clear what the type. The `location` option, I agree, should have the type

File: js/renovation/viz/core/tooltip.tsx (hunk 3 above)
Comment (Krijovnick) [reply js/renovation/viz/core/tooltip.tsx]: yes, it is copy paste from the original code :)

**Review by MaximKudriavtsev — APPROVED** [review]

**Review by roman-simionov — APPROVED** [review]

**Review by dxBeardedBear — APPROVED** [review]

**Pull request merged**